detection_rules/ecs.py

```diff
@@ -176,19 +176,22 @@ class KqlSchema2Eql(eql.Schema):
     }
 
     def __init__(self, kql_schema):
-        self.kql_schema = kql_schema
         eql.Schema.__init__(self, {}, allow_any=True, allow_generic=False, allow_missing=False)
+        self.schema = kql_schema
```

Comment:
you actually dont need any of this schema renaming stuff. This was originally done to normalize the schema field name in eql. Since this is all just handled in detection-rules, we can leave it all as is and just reference `kql_schema` where needed.

Reply:
I think it it still is necessary. Without it the schema does not return the type hint correctly in `type_hint = self._schema.schema.get(str(outer.node), "unknown")`, but I am not exactly sure why this is the case. Maybe the name is being overridden somewhere?

Reply:
I was being dumb, this issue is because self can be either `ecs.KqlSchema2Eql` or `eql.Schema` fixed now.